src/main/webapp/3Dscript.js

```diff
@@ -11,15 +11,61 @@ let controls;
 let orientation;
 let trajectory;
 let pose;
+let poseLength;
 const poseTransform = {
   /* Unit: 4 meters*/ translateX: 0,
   /* Unit: 4 meters*/ translateZ: 0,
   /* Unit: degrees*/ rotate: 0,
   /* Scalar*/ scale: 1};
+
+/* variables used to keep track of indices of trajectory wanted to be viewed. */
+const poseStartIndex = {start: 0};
+const poseEndIndex = {end: Number.MAX_VALUE};
+
+/* Last indices to be picked for boundaries of a partial trajectory. */
+let oldStart=-1;
+let oldEnd;
+
 const apiKey = 'AIzaSyDCgKca9sLuoQ9xQDfHUvZf1_KAv06SoTU';
 
+/* Matrices used to scale and unscale pose cylinders */
+const scaleCylinder = 2;
+const scaleMatrix = new THREE.Matrix4().makeScale(
+    scaleCylinder, scaleCylinder, scaleCylinder );
+const scaleInverseMatrix = new THREE.Matrix4().makeScale(
+    1/scaleCylinder, 1/scaleCylinder, 1/scaleCylinder );
+
+/* Index of last selected point to view. */
+let selectedIndex= -1;
+
+/* For hiding trajectory. */
+const scaleHide = 400;
+const zeroMatrix = new THREE.Matrix4().makeScale(
+    1/scaleHide, 1/scaleHide, 1/scaleHide );
+const nonZeroMatrix = new THREE.Matrix4().makeScale(
+    scaleHide, scaleHide, scaleHide);
+
+/* Temp matrices used for matrix multiplication. */
+const instanceMatrix = new THREE.Matrix4();
+const matrix = new THREE.Matrix4();
```

Comment:
Looks like these variables are only used within `updateMatrixAtIndex`. Can they be local to that function?